packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3536,11 +3553,67 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     }
   }
 
+  final bool _platformSupportsFadeOnScroll = defaultTargetPlatform == TargetPlatform.android
+                                          || defaultTargetPlatform == TargetPlatform.iOS;
+
   void _onEditableScroll() {
-    _selectionOverlay?.updateForScroll();
+    if (!_platformSupportsFadeOnScroll || _webContextMenuEnabled) {
+      _selectionOverlay?.updateForScroll();
+    }
     _scribbleCacheKey = null;
   }
 
+  void _handleContextMenuOnParentScroll(ScrollNotification notification) {
+    if (_internalScrolling) {
+      return;
+    }
+    _handleContextMenuOnScroll(notification);
+  }
+
+  void _handleContextMenuOnScroll(ScrollNotification notification) {
+    if (!_platformSupportsFadeOnScroll || _webContextMenuEnabled) {
+      return;
+    }
+    // When the scroll begins and the toolbar is visible, hide it
+    // until scrolling ends.
+    //
+    // When scrolling ends and the toolbar was present at the
+    // beginning of the scroll we should re-show the toolbar only
+    // if the selection endpoints are within the viewport at the
+    // end of the scroll. If not then we should schedule the
+    // toolbar to be shown when we scroll the selection back into
+    // view.
+    //
+    // A scheduled toolbar will only be shown if the editing state
+    // did not change between the time the toolbar was first
+    // scheduled to be shown to when it is ready to be shown.
+    if (notification is ScrollStartNotification) {
+      _toolbarVisibleAtScrollStart = _selectionOverlay != null && _selectionOverlay!.toolbarIsVisible;
+      if (_toolbarVisibleAtScrollStart) {
+        hideToolbar(false);
+      }
+    } else if (notification is ScrollEndNotification) {
+      final bool selectionIsVisible = renderEditable.selectionStartInViewport.value || renderEditable.selectionEndInViewport.value;
+      final bool renderEditableInView = renderEditable.getTransformTo(null) != Matrix4.zero();
+      if ((_toolbarVisibleAtScrollStart || _showToolbarOnScreenScheduled) && (selectionIsVisible && renderEditableInView)) {
+        _toolbarVisibleAtScrollStart = false;
+        if (_showToolbarOnScreenScheduled) {
+          _showToolbarOnScreenScheduled = false;
+          if (_valueWhenShowToolbarOnScreenScheduled == _value) {
+            showToolbar();
+          }
+          _valueWhenShowToolbarOnScreenScheduled = null;
+        } else {
+          showToolbar();
+        }
+      } else if (_toolbarVisibleAtScrollStart) {
+        _showToolbarOnScreenScheduled = true;
+        _valueWhenShowToolbarOnScreenScheduled = _value;
```

Comment:
You don't need the boolean since the extra bit can be represented by setting _valueWhenShowToolbarOnScreenScheduled to null?

Reply:
Also this is a bit hard to read.

Reply:
That makes sense. Thanks for the suggestion, should look cleaner now.